sysroot/usr/include/kernel/lib/mutex.hpp

```diff
@@ -8,25 +8,29 @@
  * @copyright Copyright the Panix Contributors (c) 2020
  * 
  */
-
+#ifndef PANIX_MUTEX_HPP
+#define PANIX_MUTEX_HPP
```

Comment:
Why not use `#pragma once`?

Reply:
We didn't know about it when we started. I'm all for it now though.

Reply:
Personally, I prefer this style over pragma once, but I'll concede my preference if everyone else wants pragma.

Reply:
I don't think there is too great of an argument to use this style over `#pragma once` unless you are using it as documentation for what file you are in, but there should be a top of the file docstring doing so.

Generally, it seems the standard for modern code is to use `#pragma once` as it is less error-prone and pollutes the preprocessor scope less.
https://docs.microsoft.com/en-us/cpp/preprocessor/once?view=msvc-160
https://stackoverflow.com/questions/787533/is-pragma-once-a-safe-include-guard

Reply:
I didn't realize there were actual differences / benefits to using pragma once. If that's the case then I'm alright with using it.

Reply:
Apparently GCC optimizes include guards to act almost identically to pragma once, but we can still make the switch.

Reply:
See #174 and its associated MR.